plash_cli/core.py

```diff
@@ -106,21 +106,58 @@ def login(
     else: print("Authentication timed out.")
 
 # %% ../nbs/00_core.ipynb 15
+pat = r'(?m)^# /// (?P<type>[a-zA-Z0-9-]+)$\s(?P<content>(^#(| .*)$\s)+)^# ///$'
+
+def _deps(script: bytes | str) -> dict | None:
+    '''Get the dependencies from the script. From: https://peps.python.org/pep-0723/'''
+    name = 'script'
+    if isinstance(script, bytes): script = script.decode('utf-8')
+    matches = L(re.finditer(pat, script)).filter(lambda m: m.group('type') == name)
+    if len(matches) > 1: raise ValueError(f'Multiple {name} blocks found')
+    elif len(matches) == 1:
+        content = ''.join(line[2:] if line.startswith('# ') else line[1:]
+                          for line in matches[0].group('content').splitlines(keepends=True))
+        return '\n'.join(tomllib.loads(content)['dependencies'])
+    else: return None
+
+# %% ../nbs/00_core.ipynb 16
+def _tarz(path)->io.BytesIO: # Buffer of tar directory
+    "Creates a tar archive of a directory, excluding files based on is_included"
+    tarz = io.BytesIO()
+    files = L(path if path.is_file() else Path(path).iterdir()).filter(is_included)
+
+    with tarfile.open(fileobj=tarz, mode='w:gz') as tar:
+        for f in files: tar.add(f, arcname=f.name)
+        if path.is_file():
+            if deps := _deps(path.read_bytes()):
+                info = tarfile.TarInfo('requirements.txt')
+                info.size = len(deps)
+                tar.addfile(info, io.BytesIO(deps.encode('utf-8')))
+    tarz.seek(0)
+    return tarz, len(files)
```

Comment:
So far none of the functions using _tarz use the len(files) output. Perhaps we can remove it?